### PR TITLE
Shopping page

### PR DESCRIPTION
**Pull request by konevm**
Create cart and shopping page, create wrapper as HOC for modals

### CODE REVIEW

File: src/components/Routes/routesEnum.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export enum Routes {
```
Comment (bobkov-evgeny) [inline src/components/Routes/routesEnum.ts:1]: Let's create new folder at /src and name it "constants", inside this folder let's create new file app-routes.constants.ts. Inside put your enum, just rename it to "AppRoutes". 

And please keep in mind enum naming convention example: 
var WinnerEnum = {
    PLAYER_1: 1,
    PLAYER_2: 2,
    DRAW: 0
};

CamelCase for enum name, and UPPERCASE for enum fields

File: src/App.tsx [hunk 2]
```diff
@@ -1,15 +1,21 @@
 import React from "react";
 import { Routes, Route, Navigate } from "react-router-dom";
 import { Layout } from "./components/Layout/Layout";
+import { Routes as routes } from "./components/Routes/routesEnum";
```
Comment (bobkov-evgeny) [inline src/App.tsx:4]: Don't really understand why we need this "as" operator )

File: src/Pages/Cart/Cart.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { useAppSelector } from "../../app/hooks";
+import { Link, useNavigate } from "react-router-dom";
+import { Routes } from "../../components/Routes/routesEnum";
+import "./Cart.scss";
```
Comment (bobkov-evgeny) [inline src/Pages/Cart/Cart.tsx:5]: Please check your imports

File: src/Pages/Cart/Cart.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+};
+
+const Cart: React.FC = () => {
+  const productsLink = `../${Routes.products}`;
+  const { cart, totalPrice, showBurgerMenu } = useAppSelector((store) => store.data);
+  const navigate = useNavigate();
+  return (
+    <div className="app__cart">
+      {cart.length === 0 ? "Your cart is empty." : "Your products:"}
+      {cart.length === 0 && <Link to={productsLink}> Continue shopping </Link>}
```
Comment (bobkov-evgeny) [inline src/Pages/Cart/Cart.tsx:20]: Can't we just pass here Routes.products variable?

File: src/Pages/MainPage/MainPage.tsx [hunk 5]
```diff
@@ -10,7 +10,7 @@ const MainPage: React.FC = () => {
   return (
     <div className="app__main-page">
       <div className="mainPage__banner">
-        <Link to="/new" className="banner__ad">
+        <Link to="new" className="banner__ad">
```
Comment (bobkov-evgeny) [inline src/Pages/MainPage/MainPage.tsx:13]: Forgot to use routes enum?

File: src/Pages/Shop/Shop.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+interface IShopList {
+  products: IProduct[];
+  new?: boolean;
+}
+interface ISortedProps {
+  new?: boolean;
+}
+
+const List: React.FC<IShopList> = ({ products, new: isNew }) => {
```
Comment (bobkov-evgeny) [inline src/Pages/Shop/Shop.tsx:17]: I think we simply can rename this prop to isNew (in interface as well)

File: src/components/CartItem/CartItem.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import ArrowForwardIosIcon from "@mui/icons-material/ArrowForwardIos";
+import "./CartItem.scss";
+
+interface IItemProps {
+  product: ICartItem;
+}
+
+const CartItem: React.FC<IItemProps> = ({ product }) => {
+  const dispatch = useAppDispatch();
+  const { name, image, price, amount } = product;
```
Comment (bobkov-evgeny) [inline src/components/CartItem/CartItem.tsx:15]: To my mind, it's better to rename "image" to "imageUrl"

File: src/components/CartItem/CartItem.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  return (
+    <div className="cart__item">
+      <div className="item__image" style={{ backgroundImage: `url(${image})` }}></div>
```
Comment (bobkov-evgeny) [inline src/components/CartItem/CartItem.tsx:18]: We have empty div, let's close it />

File: src/components/EcoProducts/EcoProducts.tsx [hunk 9]
```diff
@@ -15,7 +15,7 @@ const EcoProducts: React.FC = () => {
             {item}
           </li>
         ))}
-        <Link className="info__link" to="/shop">
+        <Link className="info__link" to="products">
```
Comment (bobkov-evgeny) [inline src/components/EcoProducts/EcoProducts.tsx:18]: AppRoutes enum?

File: src/components/Glowing/Glowing.tsx [hunk 10]
```diff
@@ -12,7 +12,7 @@ const Glowing: React.FC = () => {
           nutrient rich natural ingredients to elevate your skincare routine and give you glowing
           skin. Because the best makeup is NO makeup.
         </p>
-        <Link className="glowing__link" to="/shop">
+        <Link className="glowing__link" to="products">
```
Comment (bobkov-evgeny) [inline src/components/Glowing/Glowing.tsx:15]: The same -  AppRoutes enum?

File: src/components/Header/Header.tsx [hunk 11]
```diff
@@ -19,7 +23,7 @@ const Header: React.FC = () => {
           dispatch(changeBurgerMenuVisibility());
         }}
       />
-      <img src={Logo} alt="KAL HANS" className="header__logo" />
+      <Link to={Routes.main} className="header__logo" style={{ backgroundImage: `url(${Logo})` }} />
```
Comment (bobkov-evgeny) [inline src/components/Header/Header.tsx:26]: If "Logo" is also URL string, let's rename it to logoUrl

**Review by bobkov-evgeny — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 2 above)
Comment (konevm) [reply src/App.tsx:4]: Routes already exists in react-router-dom

File: src/components/Routes/routesEnum.ts (hunk 1 above)
Comment (konevm) [reply src/components/Routes/routesEnum.ts:1]: done

File: src/Pages/Cart/Cart.tsx (hunk 4 above)
Comment (konevm) [reply src/Pages/Cart/Cart.tsx:20]: i don't want to write `../${AppRoutes.PRODUCTS}` twice

File: src/Pages/MainPage/MainPage.tsx (hunk 5 above)
Comment (konevm) [reply src/Pages/MainPage/MainPage.tsx:13]: "new" is rewrite soon

File: src/components/EcoProducts/EcoProducts.tsx (hunk 9 above)
Comment (konevm) [reply src/components/EcoProducts/EcoProducts.tsx:18]: done

File: src/components/Glowing/Glowing.tsx (hunk 10 above)
Comment (konevm) [reply src/components/Glowing/Glowing.tsx:15]: done

File: src/components/Header/Header.tsx (hunk 11 above)
Comment (konevm) [reply src/components/Header/Header.tsx:26]: done

File: src/components/CartItem/CartItem.tsx (hunk 8 above)
Comment (konevm) [reply src/components/CartItem/CartItem.tsx:18]: done

File: src/components/CartItem/CartItem.tsx (hunk 7 above)
Comment (konevm) [reply src/components/CartItem/CartItem.tsx:15]: done

File: src/Pages/Shop/Shop.tsx (hunk 6 above)
Comment (konevm) [reply src/Pages/Shop/Shop.tsx:17]: done

**Review by bobkov-evgeny — APPROVED** [review]

**Pull request merged**